lib/rules/render-result-naming-convention.ts

```diff
@@ -30,15 +35,33 @@ export default createTestingLibraryRule<Options, MessageIds>({
   defaultOptions: [],
 
   create(context, _, helpers) {
+    const renderWrapperNames: string[] = [];
+
+    function detectRenderWrapper(node: TSESTree.Identifier): void {
+      const innerFunction = getInnermostReturningFunction(context, node);
+
+      if (innerFunction) {
+        renderWrapperNames.push(getFunctionName(innerFunction));
+      }
+    }
+
     return {
+      'CallExpression Identifier'(node: TSESTree.Identifier) {
+        if (helpers.isRenderUtil(node)) {
+          detectRenderWrapper(node);
+        }
+      },
       VariableDeclarator(node) {
         const initIdentifierNode = getDeepestIdentifierNode(node.init);
 
         if (!initIdentifierNode) {
           return;
         }
 
-        if (!helpers.isRenderUtil(initIdentifierNode)) {
+        if (
+          !helpers.isRenderUtil(initIdentifierNode) &&
+          !renderWrapperNames.includes(initIdentifierNode.name)
```

Comment:
Detecting nested/wrapped `render` usages was as simple as adding these few lines of code 😨